website/docs/en/config/optimization.mdx

```diff
@@ -285,50 +285,60 @@ See available options for configuring this behavior in the [SplitChunksPlugin](/
   ]}
 />
 
-```json
-{
-  "name": "npm module",
-  "version": "1.0.0",
-  "sideEffects": ["**/src/*.js"]
-}
-```
-
-Tells Rspack to recognise the sideEffects flag in package.json or rules to skip over modules which are flagged to contain no side effects when exports are not used.
-
-:::tip
-Please note that `sideEffects` should be in the npm module's `package.json` file and doesn't mean that you need to set `sideEffects` to `false` in your own project's `package.json` which requires that npm module.
-:::
-
-`optimization.sideEffects` depends on [`optimization.providedExports`](#optimizationprovidedexports) to be enabled.
-This dependency has a build time cost, but eliminating modules has positive impact on performance because of less code generation.
-Effect of this optimization depends on your codebase, try it for possible performance wins.
+If you only want Rspack use the manual `sideEffects` flag via (`package.json` and `module.rule.sideEffects`) and don't analyse source code:
 
 ```js title="rspack.config.mjs"
 export default {
   //...
   optimization: {
-    sideEffects: true,
+    sideEffects: 'flag',
   },
 };
 ```
 
-If you only want Rspack use the manual `sideEffects` flag via (`package.json` and `module.rule.sideEffects`) and don't analyse source code:
+`flag` tells Rspack to recognise the sideEffects flag in package.json or rules to skip over modules which are flagged to contain no side effects when exports are not used.
+
+`true` tells Rspack not only recognise the sideEffects flag, but also analyse modules which are not flagged explicitly, and determine if they have side effects or not.
 
 ```js title="rspack.config.mjs"
 export default {
   //...
   optimization: {
-    sideEffects: 'flag',
+    sideEffects: true,
   },
 };
 ```
 
-The `'flag'` value is used by default in non-production builds.
-
 :::tip
-When `optimization.sideEffects` is true , Rspack will also flag modules as side effect free when they contain only side effect free statements.
+`optimization.sideEffects` depends on [`optimization.providedExports`](#optimizationprovidedexports) to be enabled.
+This dependency has a build time cost, but eliminating modules has positive impact on performance because of less code generation.
+Effect of this optimization depends on your codebase, try it for possible performance wins.
 :::
 
+In `package.json`, you can use true or false to mark whether all modules under the current package contain side effects.
```

Comment:
Can we move the `package.json` part to https://rspack.dev/guide/optimization/tree-shaking?

I think some users will want to refer to this guide even when they don't actively configure the `optimization.sideEffects` option.

Reply:
There is already a ref at tree-shaking doc which guide user to `sideEffects` option, [here](https://github.com/web-infra-dev/rspack/blob/1b4c7b3a13590b8d18dd2e70ebd9f47c753efa3f/website/docs/en/guide/optimization/tree-shaking.mdx#side-effects-analysis)

Reply:
The current problem with Rspack's documentation is that the configuration page is very long and the guide documentation is lacking (compared to webpack)

In the future, we will try to organize the content like this:

- Provide a separate guide page for each major feature
- If a configuration item has a corresponding guide, put the detailed content in the guide and link it